build-scripts/createRelease.ts

```diff
@@ -0,0 +1,62 @@
+/*!
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as child_process from 'child_process'
+import * as fs from 'fs-extra'
+import * as path from 'path'
+
+const changesDirectory = '.changes'
+const nextReleaseDirectory = path.join(changesDirectory, 'next-release')
+// tslint:disable-next-line:no-var-requires no-unsafe-any
+const releaseVersion = require(path.join('..', 'package.json')).version
+const changesFile = path.join(changesDirectory, `${releaseVersion}.json`)
+
+fs.mkdirpSync(nextReleaseDirectory)
+
+const changeFiles = fs.readdirSync(nextReleaseDirectory)
+if (changeFiles.length === 0) {
+    console.log('Error! no changes to release!')
+    process.exit(-1)
+}
+try {
+    fs.accessSync(changesFile)
+    console.log(`Error! changelog file ${changesFile} already exists for version ${releaseVersion}!`)
+    process.exit(-1)
+} catch (err) {
+    // This is what we want to happen, the file should not exist
+}
+
+const now = new Date()
+const timestamp = `${now.getFullYear()}-${now.getMonth()}-${now.getDay()}`
+const changelog: any = {
+    date: timestamp,
+    version: releaseVersion,
+    entries: []
+}
+
+for (const changeFile of changeFiles) {
+    const file = JSON.parse(fs.readFileSync(path.join(nextReleaseDirectory, changeFile)).toString())
```

Comment:
How is perf for this script with a large number of changeFiles? Would it make a noticeable difference to switch to async I/O and process files in parallel? I.e.

```typescript
const readFile = util.promisify(fs.readFile)
...
changelog.entries = await Promise.all(changeFiles.map(changeFile =>
    readFile(path.join(nextReleaseDirectory, changeFile)).then(content => JSON.parse(content.toString())
))
```

Reply:
If we need to process thousands of changefiles, we're in a pretty poor state.
I think the usage scenario of this script lends itself towards simpler code that is easier to manage than realizing performance in highly uncommon situations.

Reply:
Perf does not matter for this script, even with thousands of changelog files we would be fine.